inbm/dispatcher-agent/dispatcher/schedule/apscheduler.py

```diff
@@ -0,0 +1,105 @@
+"""
+    Uses APScheduler to execute scheduled tasks.
+
+    Copyright (C) 2024 Intel Corporation
+    SPDX-License-Identifier: Apache-2.0
+"""
+
+import logging
+from typing import Callable, Optional
+from datetime import datetime
+
+from .schedules import Schedule, SingleSchedule, RepeatedSchedule
+from apscheduler.schedulers.background import BackgroundScheduler
+from .sqlite_manager import SqliteManager
+from ..constants import SCHEDULED
+
+logger = logging.getLogger(__name__)
+
+
+class APScheduler:
+    def __init__(self, sqlite_mgr: SqliteManager) -> None:
+        self._scheduler = BackgroundScheduler()
+        self._sqlite_mgr = sqlite_mgr
+
+    def start(self) -> None:
+        """Start the scheduler"""
+        logger.debug("")
+        self._scheduler.start()
+
+    def add_single_schedule_job(self, callback: Callable, single_schedule: SingleSchedule) -> None:
+        """Add the job for single schedule.
+
+        @param callback: The function to be called.
+        @param single_schedule: SingleSchedule object
+        """
+        logger.debug("")
+        if self.is_schedulable(single_schedule):
+            self._sqlite_mgr.update_status_in_database(single_schedule, SCHEDULED)
+            for manifest in single_schedule.manifests:
+                self._scheduler.add_job(callback, 'date', run_date=single_schedule.start_time, args=[manifest])
+
+    def add_repeated_schedule_job(self, callback: Callable, repeated_schedule: RepeatedSchedule) -> None:
+        """Add the job for repeated schedule.
+
+        @param callback: The function to be called.
+        @param repeated_schedule: RepeatedSchedule object.
+        """
+        logger.debug("")
+        for manifest in repeated_schedule.manifests:
+            logger.debug("To be implemented.")
+
+    def is_schedulable(self, schedule: Schedule) -> bool:
+        """Check if the schedule can be scheduled.
```

Comment:
We shouldn't need this method since the parameter type will always ensure it's a type of schedule.